mutate: improve the documentation (#974)
joakim-brannstrom/dextool: plugin/mutate/README.md

# Dextool Mutate

Dextool's plugin for mutation testing of C/C++ projects. It can help you design
new tests and evaluate the quality of existing tests by measuring their ability
to detect artificially injected faults.

## Features

* Provides support for conventional mutation operators: AOR, ROR, DCC, DCR,
  LCR, COR, SDL. See [here for details](https://github.com/joakim-brannstrom/dextool/blob/master/plugin/mutate/doc/design/mutations.md).

> * [Can continue from where a testing session was interrupted](#incremental-mutation-test).

Reviewer: ```suggestion
* Can [continue](#incremental-mutation-test) from where a testing session was interrupted.
```

> * [Allows multiple instances to be run in parallel](#parallel-run).

Reviewer: ```suggestion
* Allows multiple instances to be [run in parallel](#parallel-run).
```

* Makes type aware ROR to reduce the number of equivalent mutants.
* Can reuse previous results when a subset of the SUT changes by only testing those changes (files for now).
* Provides multiple report formats (Markdown, compiler warnings, JSON, HTML).
* Lets a user modify it by using a SQLite database as intermediary storage.

> * [Can automatically rerun the mutations that previously survived when new tests are added to the test suite](#re-test-alive).

Reviewer: ```suggestion
* Can automatically [rerun the mutations that previously survived](#re-test-alive) when new tests are added to the test suite.
```

* Does automatic handling of infinite loops (timeout).
* Works with all C++ versions.
* Works with C++ templates.
* Has a simple workflow.
* Integrates without modifications to the projects build system.

> * Detect test cases that do not kill any mutants. These test cases most *probably* do not verify anything.

Reviewer: ```suggestion
* Detects "useless" test cases that do not kill any mutants.
```

> * Detect test cases that kill the same mutants. These is an indication of redundant test cases.

Reviewer: ```suggestion
* Detects "redundant" test cases that kill the same mutants.
```

> * Present an analyze of the test cases and what is unique about them. A test case that has no unique killes is an indication that it is redundant.

Reviewer: ```suggestion
* Detects "redundant" test cases that do not uniquely kill any mutants.
```
